sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3218,6 +3218,8 @@ class Dataset[T] private[sql](
         fr.inputFiles
       case r: HiveTableRelation =>
         r.tableMeta.storage.locationUri.map(_.toString).toArray
+      case DataSourceV2ScanRelation(table: FileTable, _, _) =>
+        table.fileIndex.inputFiles
       case DataSourceV2Relation(table: FileTable, _, _) =>
```

Comment:
In a previous discussion, we decided to make `V2ScanRelationPushDown` mandatory, so that `DataSourceV2Relation` won't appear in the optimized plan. Why do we change mind?